File: Telerik.StarterKit.Widgets.News/Resources/Scripts/RelatedNewsDesigner.js
/// <reference name="MicrosoftAjax.js"/>
/// <reference name="Telerik.Sitefinity.Resources.Scripts.jquery-1.3.2.min-vsdoc2.js" assembly="Telerik.Sitefinity.Resources"/>
Type.registerNamespace("Telerik.StarterKit.Widgets.News");

// ------------------------------------------------------------------------
// ContentBlockDesigner class
// ------------------------------------------------------------------------
Telerik.StarterKit.Widgets.News.RelatedNewsDesigner = function (element) {
    this._propertyEditor = null;
    this._numberOfNewsToShowEditor = null;
    Telerik.StarterKit.Widgets.News.RelatedNewsDesigner.initializeBase(this, [element]);
}

Telerik.StarterKit.Widgets.News.RelatedNewsDesigner.prototype = {

    /* ************************* set up and tear down ************************* */
    initialize: function () {
        Telerik.StarterKit.Widgets.News.RelatedNewsDesigner.callBaseMethod(this, 'initialize');
    },
    dispose: function () {
        Telerik.StarterKit.Widgets.News.RelatedNewsDesigner.callBaseMethod(this, 'dispose');
    },

    /* ************************* public methods ************************* */
    applyChanges: function () {
//        this.get_propertyEditor().get_control().Value = this._numberOfNewsToShowEditor.get_value();
        var controlData = this.get_propertyEditor().get_control();
        controlData.NumberOfNewsToShow = $("#numberOfNewsToShowEditor").val();
    },

    refreshUI: function () {
        var data = this.get_propertyEditor().get_control();
        $("#numberOfNewsToShowEditor").val(data.NumberOfNewsToShow);

//        var value = this.get_propertyEditor().get_control();
//        if (value) {
//            this._numberOfNewsToShowEditor.set_value(value);
//        }
    },

    /* ************************* properties ************************* */
    // gets the reference to the propertyEditor control
    get_propertyEditor: function () {
        return this._propertyEditor;
    },
    // sets the reference fo the propertyEditor control
    set_propertyEditor: function (value) {
        this._propertyEditor = value;
    }
};
Telerik.StarterKit.Widgets.News.RelatedNewsDesigner.registerClass('Telerik.StarterKit.Widgets.News.RelatedNewsDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);

if (typeof (Sys) !== 'undefined') Sys.Application.notifyScriptLoaded();

File: Telerik.StarterKit.Widgets.News/RelatedNewsDesigner.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;
using Telerik.Sitefinity.Web.UI.Fields;
using Telerik.Sitefinity.Web.UI.ControlDesign;
using Telerik.Sitefinity.Web.UI;
using Telerik.Sitefinity.Modules.GenericContent.Web.UI;
using System.Web.UI.WebControls;

namespace Telerik.StarterKit.Widgets.News
{
    public class RelatedNewsDesigner : ControlDesignerBase
    {
        private const string layoutTemplateName = "Telerik.StarterKit.Widgets.News.Resources.Views.RelatedNewsDesignerTemplate.ascx";
        private const string designerScriptName = "Telerik.StarterKit.Widgets.News.Resources.Scripts.RelatedNewsDesigner.js";

        #region Properties

        /// <summary>
        /// Gets the name of the embedded layout template.
        /// </summary>
        /// <value></value>
        /// <remarks>
        /// Override this property to change the embedded template to be used with the dialog
        /// </remarks>
        protected override string LayoutTemplateName
        {
            get
            {
                return layoutTemplateName;
            }
        }

        #endregion

        //#region Control references

        ///// <summary>
        ///// Gets the text control containing the content HTML.
        ///// </summary>
        ///// <value>The instance of the text control.</value>
        //protected virtual TextBox NumberOfNewsToShowEditor
        //{
        //    get
        //    {
        //        return this.Container.GetControl<TextBox>("numberOfNewsToShowEditor", true);
        //    }
        //}

        //#endregion

        #region Overridden methods

        /// <summary>
        /// Initializes the controls.
        /// </summary>
        /// <param name="container"></param>
        /// <remarks>
        /// Initialize your controls in this method. Do not override CreateChildControls method.
        /// </remarks>
        protected override void InitializeControls(GenericContainer container)
        {
            base.DesignerMode = ControlDesignerModes.Simple;

            //var contentBlock = (RelatedNews)this.PropertyEditor.Control;
            //this.NumberOfNewsToShowEditor.Text = contentBlock.NumberOfNewsToShow.ToString();
            //this.HtmlEditor.Value = LinkParser.ResolveLinks(contentBlock.Html, DynamicLinksParser.GetContentUrl, null, true);
        }

        /// <summary>
        /// Gets the script descriptors.
        /// </summary>
        /// <returns></returns>
        public override IEnumerable<ScriptDescriptor> GetScriptDescriptors()
        {
            var scriptDescriptors = new List<ScriptDescriptor>(base.GetScriptDescriptors());
            //var descriptor = (ScriptControlDescriptor)scriptDescriptors.Last();
            //descriptor.AddComponentProperty("numberOfNewsToShowEditor", this.NumberOfNewsToShowEditor.ClientID);
            //scriptDescriptors.Add(descriptor);
            return scriptDescriptors.ToArray();
        }

        /// <summary>
        /// Gets the script references.
        /// </summary>
        /// <returns></returns>
        public override IEnumerable<ScriptReference> GetScriptReferences()
        {
            string assembly = typeof(RelatedNewsDesigner).Assembly.FullName;
            var scripts = new List<ScriptReference>(base.GetScriptReferences());
            scripts.Add(new ScriptReference(RelatedNewsDesigner.designerScriptName, assembly));
            return scripts.ToArray();
        }

        protected override HtmlTextWriterTag TagKey
        {
            get
            {
                //we use div wrapper tag to make easier common styling
                return HtmlTextWriterTag.Div;
            }
        }

        #endregion
    }
}
